crypto/src/conversation/public_group_state.rs

```diff
@@ -0,0 +1,138 @@
+use std::io::{Read, Write};
+
+use openmls::prelude::{PublicGroupState, Verifiable, VerifiablePublicGroupState};
+use tls_codec::{Error, TlsDeserialize, TlsSerialize, TlsSize};
+
+use crate::{CryptoResult, MlsError};
+
+/// # PublicGroupStateEncryption
+///
+/// In order to guarantee confidentiality of the [PublicGroupState] on the wire a domain can
+/// request it to be encrypted when sent to the Delivery Service.
+///
+/// ```text
+/// enum {
+///     reserved(0),
+///     unencrypted(1),
+///     jwe_encrypted(2),
+///     (255)
+/// } PublicGroupStateEncryption;
+/// ```
+#[derive(Debug, Clone, Eq, PartialEq, TlsSerialize, TlsDeserialize, TlsSize)]
+#[repr(u8)]
+pub enum PublicGroupStateEncryption {
+    Reserved = 0,
+    Unencrypted = 1,
+    JweEncrypted = 2,
+}
+
+/// # RatchetTreeType
+///
+/// In order to spare some precious bytes, a [PublicGroupState] can have different representations.
+///
+/// ```text
+/// enum {
+///     reserved(0),
+///     full(1),
+///     delta(2),
+///     by_ref(3),
+///     (255)
+/// } RatchetTreeType;
+/// ```
+#[derive(Debug, Clone, Eq, PartialEq, TlsSerialize, TlsDeserialize, TlsSize)]
+#[repr(u8)]
+pub enum RatchetTreeType {
+    Reserved = 0,
+    /// Plain old and complete [PublicGroupState]
+    Full = 1,
+    /// Contains [PublicGroupState] changes since previous epoch (Not yet implemented)
+    /// (see [draft](https://github.com/rohan-wire/ietf-drafts/blob/main/mahy-mls-ratchet-tree-delta/draft-mahy-mls-ratchet-tree-delta.md))
+    Delta = 2,
+    ByRef = 3,
+}
+
+/// Represents the byte array in [PublicGroupStateBundle]
+#[derive(Debug)]
+#[non_exhaustive]
```

Comment:
Do we really need this? This is likely to be used only internally in the crate and all our consumers don't use rust directly so it will make no difference for them.

Reply:
good catch